packages/block-editor/src/components/url-input/index.js

```diff
@@ -173,6 +172,7 @@ class URLInput extends Component {
 				this.setState( {
 					suggestions,
 					loading: false,
+					showSuggestions: !! suggestions.length,
```

Comment:
There are a few places that check for both `showSuggestions && !! suggestions.length`. If `showSuggestions` is now tied to this, maybe we should modify those checks as well. Here are two I found:

https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/url-input/index.js#L453-L454

https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/url-input/index.js#L468-L469

Reply:
I'd rather not touch too much about this component in this PR. I'm thinking of raising another followup Issue to track the potential need to remove `showSuggestions` entirely. It's effectively derived state and that is somewhat of an anti-pattern. We should be able to use a check of `suggestions.length` to determine whether or not to show the DOM elements. What do you think?

Reply:
Sounds good!

Reply:
See https://github.com/WordPress/gutenberg/issues/22862